### PR TITLE
DOC-10068 - Fixes for PHP transactions doc and examples

### PR DESCRIPTION
**Pull request by maria-robobug**
https://issues.couchbase.com/browse/DOC-10068

### CODE REVIEW

File: modules/howtos/examples/transactions-example.php [hunk 1]
```diff
@@ -141,85 +147,106 @@ function (TransactionAttemptContext $ctx) use ($collection) {
...
   // end::insert[]
 }
 
-function queryExamples($cluster) {
+function queryExamples($cluster)
+{
+  echo "\nRunning: queryExamplesSelect\n";
+  // TODO: scopeQualifier/scopeName options are both deprecated.
+  // https://github.com/couchbase/couchbase-php-client/blob/4.0.0/Couchbase/TransactionQueryOptions.php#L296-L325
+  // It's unclear what we should use to demonstrate scope level queries.
```
Comment (maria-robobug) [inline modules/howtos/examples/transactions-example.php:239]: https://github.com/couchbase/couchbase-php-client/blob/4.0.0/Couchbase/TransactionQueryOptions.php#L296-L325 

@avsej and @programmatix It looks like the `scopeQualifier` and  `scopeName` TransactionQueryOptions are both deprecated, but in other SDKs this is not the case.
For example in Node.js we have the `scope` option (https://docs.couchbase.com/sdk-api/couchbase-node-client/interfaces/TransactionQueryOptions.html#scope)
with no mention of deprecation, so I'm a bit confused.

For PHP SDK, is there a plan for an alternative `scope` option that isn't deprecated in future? Or is this a bug maybe?

File: modules/howtos/examples/transactions-example.php [hunk 2]
```diff
@@ -255,92 +284,110 @@ function (TransactionAttemptContext $ctx) use ($hotelChain, $country) {
...
+//         $ctx->query($bulkLoadStatement);
+//       }
+//     );
+//   } catch (\Couchbase\Exception\TransactionFailedException $e) {
+//     echo "Transaction did not reach commit point\n";
+//   } catch (\Couchbase\Exception\TransactionCommitAmbiguousException $e) {
+//     echo "Transaction possibly committed\n";
+//   }
+//   // end::querySingle[]
+// }
```
Comment (maria-robobug) [inline modules/howtos/examples/transactions-example.php:390]: @avsej  and @programmatix are **single query transactions** available for PHP?
I am unclear how we show this in PHP, from what I can see in Java (using as reference), this looks a little different?

https://docs.couchbase.com/java-sdk/current/howtos/distributed-acid-transactions-from-the-sdk.html#single-query-transactions

There seems to be a `QueryOptions.queryOptions().asTransaction()` method, but I couldn't find anything similar in PHP.

File: modules/howtos/examples/transactions-example.php [hunk 3]
```diff
@@ -56,81 +55,88 @@ function removeOrWarn($collection, $doc) {
...
-    });
-}
-catch (\Couchbase\Exception\TransactionOperationFailedException $e) {
-    echo "Transaction did not reach commit point: $e\n";
-}
-catch (\Couchbase\Exception\TransactionException $e) { 
-// TODO check is this equivalent to TransactionCommitAmbiguousError as per Node examples?
+      // Performing a SELECT N1QL query:
+      $selectQuery = 'SELECT * FROM `travel-sample`.inventory.hotel WHERE country = $1 LIMIT 5';
+      $qr = $ctx->query(
```
Comment (maria-robobug) [inline modules/howtos/examples/transactions-example.php:119]: > "it returns the same QueryResult you are used to"
> I hope this is actually a TransactionQueryResult...

@programmatix in response to your comment in the JIRA ticket, from what I can see `ctx->query(...)` returns a `QueryResult`:
https://github.com/couchbase/couchbase-php-client/blob/4.0.0/Couchbase/TransactionAttemptContext.php#L128-L140 

@avsej maybe you can expand on this, but seems the expectation is that this should be a `TransactionQueryResult`?

Node.js SDK seems to have this: https://docs.couchbase.com/sdk-api/couchbase-node-client/classes/TransactionAttemptContext.html#query

**Comment by maria-robobug** [comment]
> Also PHP, Node, Go and Python docs do not mention custom metadata collections in the doc.  If this is supported (as hopefully it is, since everyone should have milestone 2.1 fully implemented) then please add.

@programmatix I don' think this has been implemented, or at least I couldn't see anything of the sort in PHP, Node.js or Python. 
**Go** does have it I think: https://pkg.go.dev/github.com/couchbase/gocb/v2#TransactionOptions.MetadataCollection

@avsej can you confirm?

**Review by avsej — APPROVED** [review]

**Pull request merged**